src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,227 @@
+            var sortedParams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["auth_token"] = tokens.AccessToken,
+                ["charset"] = "utf-8",
+                ["format"] = "JSON",
+                ["method"] = "alipay.user.info.share",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams, Options.GetPrivateKeyBytes()));
+
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, sortedParams!);
+
+            using var response = await Backchannel.GetAsync(address, Context.RequestAborted);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+            using var document = JsonDocument.Parse(stream);
+            var rootElement = document.RootElement;
+
+            if (!rootElement.TryGetProperty("alipay_user_info_share_response", out var mainElement))
+            {
+                string errorCode = rootElement.GetProperty("error_response").GetProperty("code").GetString() !;
+                throw new Exception($"An error (Code:{errorCode}) occurred while retrieving user information.");
+            }
+
+            if (!ValidateReturnCode(mainElement, out string code))
+            {
+                throw new Exception($"An error (Code:{code}) occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, mainElement.GetString("user_id") !, ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, mainElement);
+
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+
+            return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => FormatScope(Options.Scope);
```

Comment:
You don't need this override, it's [already what the base class does](https://github.com/dotnet/aspnetcore/blob/50c635f95f0fcbe30f30175d2884f79e693f48a7/src/Security/Authentication/OAuth/src/OAuthHandler.cs#L339-L340).

Reply:
Oh！It's careless of me.